### ISSUE TITLE
Query parameters serialization is broken in generated API client

### ISSUE THREAD

**Issue by prog13** [acacode/swagger-typescript-api#228]
Hi team!

I believe that PR #223 has introduced a regression to the generated API query parameters serialization. The issue is in [this template](https://github.com/acacode/swagger-typescript-api/blob/master/templates/base/http-clients/fetch-http-client.eta#L83):

```
protected toQueryString(rawQuery?: QueryParamsType): string {
    const query = rawQuery || {};
    const keys = Object.keys(query).filter((key) => "undefined" !== typeof query[key]);
    return keys
            .map((key) =>
                typeof query[key] === "object" && !Array.isArray(query[key])
                ? this.toQueryString(query[key] as QueryParamsType)
                : this.addArrayQueryParam(query, key),
            )
            .join("&");
}
```

Condition `typeof query[key] === "object" && !Array.isArray(query[key])` incorrectly handles primitive types: if type of `query[key]` is, for example, `number`, it will be handled by the `this.addArrayQueryParam` method, which is incorrect, and will throw an exception further [on this line](https://github.com/acacode/swagger-typescript-api/blob/master/templates/base/http-clients/fetch-http-client.eta#L75):

```
private addArrayQueryParam(query: QueryParamsType, key: string) {
    const value = query[key];
    const encodedKey = encodeURIComponent(key);
    return `${value.map((val: any) => `${encodedKey}=${encodeURIComponent(typeof val === "number" ? val : `${val}`)}`).join('&')}`;
}
```

Since type of `value` is `number`, obviously it's doesn't have `.map` method and the code will break.

**Comment by prog13** [comment]
I've opened the MR #229, hope I've got it right 🤞

**Comment by js2me** [comment]
Hello @prog13 !   
Thanks for your contribution, fix of this bug was added in 8.0.3
